x/programs/runtime/runtime.go

```diff
@@ -36,12 +40,36 @@ type runtime struct {
 	engine   wazero.Runtime
 	mod      api.Module
 	meter    Meter
-	storage  Storage
 	// functions exported by this runtime
 	mu state.Mutable
 
 	closed bool
 	log    logging.Logger
+	// db for persistent storage
+	db database.Database
+	// runtime pubkey
+	callerAddress ed25519.PublicKey
+}
+
+func (r *runtime) initProgramStorage(programBytes []byte) (int64, error) {
+	count, err := GetProgramCount(r.db)
+	if err != nil {
+		r.log.Error("failed to get program counter", zap.Error(err))
+		return 0, err
+	}
+	// increment
+	err = IncrementProgramCount(r.db)
+	if err != nil {
+		r.log.Error("failed to increment program counter", zap.Error(err))
+		return 0, err
+	}
+	// store program bytes
+	err = SetProgram(r.db, count, r.callerAddress, programBytes)
+	if err != nil {
+		r.log.Error("failed to set program", zap.Error(err))
+		return 0, err
+	}
+	return int64(count), nil
 }
 
 func (r *runtime) Create(ctx context.Context, programBytes []byte) (uint64, error) {
```

Comment:
as noted in the other review I think we should drop the `Create` method from the Runtime interface.

Reply:
TODO: future conversation